packages/runtime/src/components/builtin/Image/Image.tsx

```diff
@@ -47,23 +49,33 @@ function loadImage(src: string): Promise<HTMLImageElement> {
   })
 }
 
-function imageSizes(width?: Props['width']): string {
-  const baseDevice = DEVICES.find(device => device.maxWidth == null)
-  const baseWidth = baseDevice && width && findDeviceOverride(width, baseDevice.id)
+function imageSizes(breakpoints: Breakpoints, width?: Props['width']): string {
+  const baseDevice = breakpoints.find(breakpoint => breakpoint.maxWidth == null)
+  const baseWidth = baseDevice && width && findBreakpointOverride(breakpoints, width, baseDevice.id)
   const baseWidthSize =
     baseWidth == null || baseWidth.value.unit !== 'px' ? '100vw' : `${baseWidth.value.value}px`
 
-  return DEVICES.map(device => {
-    const override = findDeviceOverride(width, device.id)
+  return breakpoints
+    .map(breakpoint => {
+      const override = findBreakpointOverride(breakpoints, width, breakpoint.id)
 
-    if (override == null || device.maxWidth == null || override.value.unit !== 'px') return null
+      if (override == null || breakpoint.maxWidth == null || override.value.unit !== 'px') {
+        return null
+      }
 
-    return `(max-width: ${device.maxWidth}px) ${Math.min(device.maxWidth, override.value.value)}px`
-  })
+      return `(max-width: ${breakpoint.maxWidth}px) ${Math.min(
+        breakpoint.maxWidth,
+        override.value.value,
+      )}px`
+    })
     .filter((size): size is NonNullable<typeof size> => size != null)
     .reduce((sourceSizes, sourceSize) => `${sourceSize}, ${sourceSizes}`, baseWidthSize)
 }
 
+function useImageSizes(...args: DropFirst<Parameters<typeof imageSizes>>) {
```

Comment:
I think I would rather we be explicit with arguments here. This looks very fancy but I think it make things harder to follow